access/views.py

```diff
@@ -39,10 +46,125 @@ def index(request):
         })
     return render(request, 'access/ready.html', {
         "courses": courses,
-        "manager": 'gitmanager' in settings.INSTALLED_APPS,
     })
 
 
+def publish(request):
+    """
+    Move a course from store to the main folder
+    """
+    if request.method != "POST":
+        raise Http404()
+
+    if "course_id" not in request.POST:
+        return HttpResponse("Missing course_id", status=400)
+
+    course_id = request.POST["course_id"]
+
+    if not request.auth.permissions.instances.has(Permission.WRITE, id=int(course_id)):
+        return HttpResponse(status=401)
+
+    root_dir = Path(settings.COURSES_PATH)
+    store_root_dir = Path(settings.COURSE_STORE)
+    course_path = root_dir / course_id
+    store_course_path = store_root_dir / course_id
+    version_id_path = root_dir / (course_id + ".version")
+    store_version_id_path = store_root_dir / (course_id + ".version")
+
+    store_version_id = None
+    if store_version_id_path.exists():
+        with open(store_version_id_path) as f:
+            store_version_id = f.read()
+
+    if store_version_id == request.POST.get("version_id"):
+        renames([
+            (store_version_id_path, version_id_path),
+            (store_course_path, course_path),
+        ])
+
+        return HttpResponse()
+
+    version_id = None
+    if version_id_path.exists():
+        with open(version_id_path) as f:
+            version_id = f.read()
+
+    if version_id == request.POST.get("version_id"):
+        return HttpResponse()
+    else:
+        return HttpResponse("Unknown version id", status=400)
+
+
+@login_required
+def configure(request):
+    '''
+    Configure a course according to the gitmanager protocol.
+    '''
+    if request.method != "POST":
+        raise Http404()
+
+    if "publish" in request.POST and request.POST["publish"]:
+        return publish(request)
+
+    if "exercises" not in request.POST or "course_id" not in request.POST:
+        return HttpResponse("Missing exercises or course_id", status=400)
+
+    exercises = json.loads(request.POST["exercises"])
+    course_id = request.POST["course_id"]
+
+    if not request.auth.permissions.instances.has(Permission.WRITE, id=int(course_id)):
+        return HttpResponse(status=401)
+
+    root_dir = Path(settings.COURSE_STORE)
+    course_path = root_dir / course_id
+    if course_path.exists():
+        rmtree(course_path)
```

Comment:
```suggestion
    try:
        rmtree(course_path)
    except OSError:
        logger.exception(...)
        # can we continue after this or should we return from the function?
```
`shutil.rmtree` has the parameter `ignore_errors` that might be useful in this case to avoid Exceptions.